core/src/avm2/optimize.rs

```diff
@@ -251,15 +255,49 @@ pub fn optimize<'gc>(
         }
     }
 
+    // TODO: Fill out all ops, then add scope stack and stack merging, too
+    let mut state_map: HashMap<i32, Locals<'gc>> = HashMap::new();
+
     let mut stack = Stack::new();
     let mut scope_stack = Stack::new();
     let mut local_types = initial_local_types.clone();
 
     for (i, op) in code.iter_mut().enumerate() {
-        if jump_targets.contains(&(i as i32)) {
+        if let Some(sources) = jump_targets.get(&(i as i32)) {
+            // Avoid handling multiple sources for now
+            if sources.len() == 1 {
+                // We can merge the locals easily, now
+                let source_i = sources[0];
+                // Because of the linear optimizer logic, this guarantees that the jump source came before the target
```

Comment:
Will the optimizer architecture change to the point where we'd need to worry about backwards jumps at some point? Or would we be splitting up optimization passes into separate functions by then?

Reply:
Yes, we will definitely have to worry about backwards jumps (since we need to implement full state merging for the verifier/optimizer, e.g. to verify the stack and scope stack and optimize `findproperty`), but that will probably be after the verifier is merged with the optimizer, which will be a near-full rewrite of all this code.